### PR TITLE
Sendgrid PR migration

### PR DESCRIPTION
**Pull request by ameypv-rudder**
## Description of the change

> The PR [here](https://github.com/rudderlabs/rudderstack-docs/pull/229) has been converted and ported as per the latest documentation formatting and styling conventions.

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: destinations/marketing/sendgrid.mdx [hunk 1]
```diff
@@ -0,0 +1,396 @@
...
+RudderStack supports SendGrid as a destination to which you can seamlessly send your customer data.
+
+## Getting started
+
+Before configuring SendGrid as a destination in RudderStack, verify if the source platform is supported by SendGrid by referring to the table below:
+
+| Connection Mode     | Web           | Mobile     | Server           |
+| :------------------ | :------------ | :--------- | :-------------|
+| **Device mode**     | -             | -          | -             |
+| **Cloud mode**      | **Supported** | -          | **Supported** |
```
Comment (saikumarrs) [inline destinations/marketing/sendgrid.mdx:20]: Why is it not supported for "Mobile"?

Comment (abandonned) [reply destinations/marketing/sendgrid.mdx:20]: Earlier we were sending properties inside integration object and for mobile version it only supported boolean values therefore we couldn't support. But now everything is inside properties therefore we can support mobile mode as well.

File: destinations/marketing/sendgrid.mdx [hunk 2]
```diff
@@ -0,0 +1,392 @@
...
+
+As mentioned in the [**Connection settings**](#connection-settings) section above, the event names for which a `track` call is made must be specified in the RudderStack dashboard. In the sample `track` call above, `testing` is the event name that should be configured in the RudderStack dashboard. Otherwise, the call will be dropped.
+
+### `content`
+
+Each object inside `content` array must contain the `type` and `value` fields. If these fields are absent, then the object will be dropped. However, note that the call will **not** be discarded.
+
+### `attachments`
+
+Each object inside attachments array must atleast contain the `content` and `filename` fields. If these fields are absent, then the object will be dropped. However the call will **not** be discarded.
```
Comment (saikumarrs) [inline destinations/marketing/sendgrid.mdx:388]: Correct the typo, "at least"

File: destinations/marketing/sendgrid.mdx [hunk 3]
```diff
@@ -0,0 +1,392 @@
+---
```
Comment (saikumarrs) [inline destinations/marketing/sendgrid.mdx:1]: The following issues need to be addressed throughout the document and also wherever applicable.

“Subject” -> “Email Subject”
“Get mail from traits” -> “Get email ID from traits”
“Reply To Email” -> “Reply-To Email ID”
"Reply To Name" -> "Reply-To Name"
“Email” -> “Email ID”
“Mail Settings” -> “Email Settings”

**Review by saikumarrs — CHANGES_REQUESTED** [review]

File: destinations/marketing/sendgrid.mdx [hunk 4]
```diff
@@ -0,0 +1,392 @@
...
+
+As mentioned in the [**Connection settings**](#connection-settings) section above, the event names for which a `track` call is made must be specified in the RudderStack dashboard. In the sample `track` call above, `testing` is the event name that should be configured in the RudderStack dashboard. Otherwise, the call will be dropped.
+
+### `content`
+
+Each object inside `content` array must contain the `type` and `value` fields. If these fields are absent, then the object will be dropped. However, note that the call will **not** be discarded.
+
+### `attachments`
+
+Each object inside attachments array must at least contain the `content` and `filename` fields. If these fields are absent, then the object will be dropped. However the call will **not** be discarded.
```
Comment (arajguha) [inline destinations/marketing/sendgrid.mdx:388]: should we call it `event` instead of `call`?

Comment (ameypv-rudder) [reply destinations/marketing/sendgrid.mdx:388]: Sure - that makes sense. I will make the change for these instances.

File: destinations/marketing/sendgrid.mdx (hunk 3 above)
Comment (ameypv-rudder) [reply destinations/marketing/sendgrid.mdx:1]: @abandonned has incorporated these changes in the webapp and the docs are updated as well.

**Review by abandonned — APPROVED** [review]

**Pull request merged**